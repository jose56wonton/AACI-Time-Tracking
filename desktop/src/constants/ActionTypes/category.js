export const GET_CATEGORIES_REQUEST = 'get_categories_request';
export const GET_CATEGORIES_SUCCESS = 'get_categories_success';
export const GET_CATEGORIES_FAILURE = 'get_categories_failure';

export const EDIT_CATEGORIES_MODAL = 'edit_categories_modal';

export const UPDATE_CATEGORY_REQUEST = 'update_category_request';
export const UPDATE_CATEGORY_SUCCESS = 'update_category_success';
export const UPDATE_CATEGORY_FAILURE = 'update_category_failure';

export const CREATE_CATEGORY_REQUEST = 'create_category_request';
export const CREATE_CATEGORY_SUCCESS = 'create_category_success';
export const CREATE_CATEGORY_FAILURE = 'create_category_failure';

export const DELETE_CATEGORY_REQUEST = 'delete_category_request';
export const DELETE_CATEGORY_SUCCESS = 'delete_category_success';
export const DELETE_CATEGORY_FAILURE = 'delete_category_failure';