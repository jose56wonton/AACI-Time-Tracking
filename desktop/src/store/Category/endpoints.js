import { HOST } from 'constants/network';

import axios from 'axios';
const DOMAIN = 'categories';

export const getCategories = () => {
  return axios.get(`${HOST}/${DOMAIN}`);
};
