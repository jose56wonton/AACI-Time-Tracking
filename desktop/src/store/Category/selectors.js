import { createSelector } from 'reselect';

export const getCategoriesFromEntities = state => state.entities.categories;
export const getCategoriesFromResults = state => state.results.categories;

export const getAnalyzeState = state => state.analyze;

export const getAllCategories = createSelector(
  getCategoriesFromEntities,
  getCategoriesFromResults,
  (categories, results) => {
    if (!results || results.size === 0) return null;
    return results.map(categoryId => {
      return categories[categoryId];
    });
  },
);

export const getSelectedCategory = createSelector(
  getCategoriesFromEntities,
  getAnalyzeState,
  (categories, analyze) => {
    if(analyze.category === -1) 
      return {};
    else 
      return categories[analyze.category];
  }
);