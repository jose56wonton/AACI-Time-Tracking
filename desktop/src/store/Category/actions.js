import { categoryActionTypes } from 'constants/ActionTypes';

import * as endpoint from './endpoints';
import { normalize } from 'normalizr';
import * as schemas from 'store/schemas';

export const getCategories = () => {
  return async dispatch => {
    dispatch({ type: categoryActionTypes.GET_CATEGORIES_REQUEST });
    try {
      const response = await endpoint.getCategories();
      const payload = normalize(
        { categories: response.data },
        schemas.categoryArray,
      );
      return dispatch({
        type: categoryActionTypes.GET_CATEGORIES_SUCCESS,
        payload
      });
    } catch (e) {
      dispatch({
        type: categoryActionTypes.GET_CATEGORIES_FAILURE,
        payload: e
      });
    }
  };
};
