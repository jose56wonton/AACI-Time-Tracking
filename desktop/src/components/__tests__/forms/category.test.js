import React from 'react';
import { shallow } from 'enzyme';

import { Category } from 'components/forms/Category/category';
import CategoryHOC from 'components/forms/Category';

const props =  {  
  classes: {},
  isSubmitting: true,
  initialValues: {},
  resetForm: jest.fn(),
  label: 'label',
  type: 'type',
  removeCategory: jest.fn(),
  errors: {}
};

const setup = overRides => {  
  return shallow(<Category {...props} {...overRides}/>);    
};

const setupHOC = overRides => {
  return shallow(<CategoryHOC {...props} {...overRides}/>);
};

describe('Category Component', () => {  
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();    
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a delete button when type===edit', () => {
    const wrapper = setup({ type: 'edit' });
    expect(wrapper).toMatchSnapshot();
  });
  it('should call resetForm on category-reset-button', () => {
    const wrapper = setup();
    const instance = wrapper.instance();
    instance.resetForm = jest.fn();
    expect(props.resetForm).toHaveBeenCalledTimes(0);
    wrapper.find('#category-reset-button').first().simulate('click');
    expect(props.resetForm).toHaveBeenCalledTimes(1);
  });
});