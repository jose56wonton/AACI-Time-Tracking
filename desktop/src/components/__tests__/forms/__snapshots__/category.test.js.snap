// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component should render a delete button when type===edit 1`] = `
<Form>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        variant="h6"
      >
        label
      </WithStyles(Typography)>
      <WithStyles(Tooltip)
        title="Delete"
      >
        <WithStyles(IconButton)
          aria-label="Delete"
          onClick={[Function]}
        >
          <pure(DeleteIcon) />
        </WithStyles(IconButton)>
      </WithStyles(Tooltip)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <C
        component={[Function]}
        helper="normal"
        label="Type"
        margin="none"
        name="type"
        type="search"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        color="error"
        variant="button"
      />
      <div>
        <WithStyles(Button)
          color="primary"
          disabled={true}
          type="submit"
          variant="contained"
        >
          Save
        </WithStyles(Button)>
        <WithStyles(Button)
          color="secondary"
          disabled={true}
          id="category-reset-button"
          onClick={[Function]}
          variant="text"
        >
          Reset
        </WithStyles(Button)>
      </div>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Form>
`;

exports[`Category Component should render correctly 1`] = `
<Form>
  <WithStyles(Grid)
    container={true}
    spacing={24}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        variant="h6"
      >
        label
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <C
        component={[Function]}
        helper="normal"
        label="Type"
        margin="none"
        name="type"
        type="search"
      />
    </WithStyles(Grid)>
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        color="error"
        variant="button"
      />
      <div>
        <WithStyles(Button)
          color="primary"
          disabled={true}
          type="submit"
          variant="contained"
        >
          Save
        </WithStyles(Button)>
        <WithStyles(Button)
          color="secondary"
          disabled={true}
          id="category-reset-button"
          onClick={[Function]}
          variant="text"
        >
          Reset
        </WithStyles(Button)>
      </div>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Form>
`;

exports[`Category Component should render correctly withStyles 1`] = `
<Category
  classes={
    Object {
      "button": "Category-button-7",
      "error": "Category-error-6",
      "field": "Category-field-4",
      "gridContainer": "Category-gridContainer-1",
      "headerRow": "Category-headerRow-3",
      "row": "Category-row-2",
      "switchBox": "Category-switchBox-5",
    }
  }
  errors={Object {}}
  initialValues={Object {}}
  isSubmitting={true}
  label="label"
  removeCategory={[Function]}
  resetForm={[Function]}
  type="type"
/>
`;
